### DISCUSSION TITLE
The page doesn't reload when using navigate inside a redirection

### DISCUSSION THREAD

**Discussion by vincentdev15** [livewire/livewire#7618] [1. Help]
Laravel 10 / Livewire 3.3

I have this code and I always use the navigate directive.

public function destroy()
{
    $this->authorize('create', $this->training);

    $this->training->delete();

    solution 1 => $this->redirectRoute('trainings.index', true); // it works
    solution 2 => $this->redirect('trainings.index', [], true, true); // it doesn't work
}

After having deleted a training, it redirects to the trainings page but the deleted training is always visible in the trainings table with solution 2 but not with solution 1.

Another problem.

public function save()
{
    $training = $this->form->save();

    $this->redirect(route('trainings.show', compact('training')), true);
    $this->redirectRoute('trainings.show', compact('training'), true, true);
}

When I save a new training in the database, it redirects to the training show page. Then when I click in the menu on the link to load the trainings page, it has a strange behavior :

- first click on the link => the new training doesn't appear in the table
- second cick on the link => the new training appears in the table

According to me, it behaves like if, when I click on the link, it load the old state of the table (without the new training).

But if I remove the navigate directive, all works fine.

I want to use the navigate directive.

I think that it's a bug, but perhaps I just don't use the code correctly ?

Thanks a lot for your help ;).

V

**Comment by joshhanley** [comment]
We're in the process of investigating all wire:navigate issues and I have tested out your scenario. I believe this has been fixed so I'm going to close this. If you're still experiencing problems, please open a new discussion. Thanks!
